File: angular/hotelinventoryapp/src/app/rooms/rooms.ts
interface Room {
    totalRooms: number
    availableRooms: number;
    bookedRooms: number;
};

enum RoomType {
    NORMAL = "NORMAL",
    LUXURY = "LUXURY",
    TERI_AUKAT_SE_BAHAR = "TERI_AUKAT_SE_BAHAR"
};

interface RoomDetails {
    roomType: RoomType;
    amenities: string[];
    roomNumber: number;
    floor: number;
    price: number;
    photo: string;
    rating: number;
    checkinTime: Date;
    checkoutTime: Date;
}

export { Room as Roomz, RoomDetails, RoomType };

File: angular/mock/src/main/java/io/commoncoder/learning/angular/mock/models/Rooms.java
package io.commoncoder.learning.angular.mock.models;

import com.google.common.collect.ImmutableList;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
@AllArgsConstructor
public class Rooms {

    RoomType roomType;
    List<String> amenities;
    Long roomNumber;
    Long floor;
    Long price;
    String photo;
    Integer rating;
    Long checkinTime;
    Long checkoutTime;

    public enum RoomType {
        NORMAL,
        LUXURY,
        TERI_AUKAT_SE_BAHAR
    }
}
